hack/kind-install-for-capk.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+
+# Copyright 2025 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# This script installs a local Kind cluster with a local container registry and the correct files
+# mounted for using CAPK to test Cluster API.
+# The default Kind CNI is disabled because it doesn't work CAPK. Instead, Calico is used.
+# MetalLB is used as a local layer 2 load balancer to support exposing the API servers of workload
+# clusters on the local machine that's running this script.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+CALICO_VERSION=${CAPI_CALICO_VERSION:-"v3.29.1"}
```

Comment:
Question: why do we rename the variable?

We could also do

```suggestion
CAPI_CALICO_VERSION=${CAPI_CALICO_VERSION:-"v3.29.1"}
```

To use the value given or default if not set (might be just copy/paste?)

Reply:
I replicated the pattern used for `CAPI_KIND_CLUSTER_NAME` to try and be consistent. I don't have strong opinions about the var names in these scripts. Do you have a preferred naming format we should use everywhere?

Reply:
just been curious. Thanks.

Reply:
I went through the vars again and tried to make things consistent. `CAPI_` is used as a prefix for vars settable by the user. "Internal" vars don't use this prefix.